### PR TITLE
Move common scope from search api to a new file

### PR DESCRIPTION
**Pull request by stefannibrasil**
The third part of #3070.

(we commented the unit test for this for now because we are going to separate a time only for testing! but by running the api docs locally we could still get the same results as before, so apparently it didn't changed the fetched results)

### CODE REVIEW

File: app/services/typeahead_service.rb [hunk 1]
```diff
@@ -125,7 +79,8 @@ def search_profiles(search_string, limit = 5)
   def search_notes(search_string, limit = 5)
```
Comment (jywarren) [inline app/services/typeahead_service.rb]: Maybe we should add to the comment to say "and package up as a TagResult" for clarity?

**Comment by jywarren** [comment]
This is looking good - so when you add the test back in, and are ready for final review, just ping me! 

These files are getting much simpler, very nice. Keep an eye out for layers of abstraction that seem unnecessary! And keep in mind how we want to be able to tell new people about how this is organized - it should be easy for a newcomer to figure out how it's all wired together, and to make a change or addition. 

Thanks, great work!

File: app/services/typeahead_service.rb (hunk 1 above)
Comment (stefannibrasil) [reply app/services/typeahead_service.rb]: yeah, good point!

**Comment by jywarren** [comment]
Great! When you think it's ready, can you start putting the `ready` label on it? Thanks and great work!

**Comment by stefannibrasil** [comment]
great, done, thanks for the help!

**Comment by stefannibrasil** [comment]
closes #3070

**Pull request merged**